genson/src/main/java/com/owlike/genson/reflect/BeanDescriptor.java

```diff
@@ -122,6 +136,8 @@ public void deserialize(T into, ObjectReader reader, Context ctx) {
         } else {
           reader.skipValue();
         }
+      } else if (unknownPropertyHandler != null) {
+        unknownPropertyHandler.onUnknownProperty(into, propName, ctx.genson.deserialize(UNKNOWN, reader, ctx));
```

Comment:
This means that if a property handler is registered it is responsible of being able to handle missing properties for all targeted types.
Another option could be to actually implement `failOnMissingProperty` as a special UnknownPropertyHandler, and chain them. That would however complicate a bit UnknownPropertyHandler as it would need to "signal" if it did handle the missing property or not. Just food for thought.

Reply:
Because it's a special case of unknown property handling, is the main reason why I was saying it could be reimplemented that way. The advantage I see to it is, unified logic and removal of unnecessary code.

Preserving some of the compatibility could be achieved by translating the boolean `failOnMissingProperty` into a handler at the builder level (and remove it from the other places like Genson and BeanDescriptor, which are less likely to be directly used).

I agree that the chaining part is not necessary.

Reply:
I see... Yes, that would probably make sense.

We just need to document that behavior, but considering that they are already mutually exclusive, we should probably document it anyway.

Let me rework the code along those lines and I'll update the PR.